sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuSortExec.scala

```diff
@@ -45,6 +45,12 @@ class GpuSortMeta(
     parent: Option[RapidsMeta[_, _, _]],
     rule: DataFromReplacementRule)
   extends SparkPlanMeta[SortExec](sort, conf, parent, rule) {
+
+  // Uses output attributes of child plan because SortExec will not change the attributes,
+  // and we need to propagate possible type conversions on the output attributes of
+  // GpuSortAggregateExec.
+  override protected val useOutputAttributesOfChild: Boolean = true
```

Comment:
I am confused why this PR changes the behavior of `GpuSortMeta`. Can you elaborate?

Reply:
The reason of changing `GpuSortMeta` is similar to `GpuShuffleMeta`. But the change here is specialized for `SortAggregateExec` with TypedImperativeAggregate functions, which will appear when we disable the configuration:
`spark.sql.execution.useObjectHashAggregateExec`.